x-pack/plugins/ml/common/types/ml_url_generator.ts

```diff
@@ -12,10 +12,10 @@ import { DataFrameAnalyticsType } from './data_frame_analytics';
 type OptionalPageState = object | undefined;
 
 export type MLPageState<PageType, PageState> = PageState extends OptionalPageState
-  ? { page: PageType; pageState?: PageState }
+  ? { page: PageType; pageState?: PageState; excludeBasePath?: boolean }
   : PageState extends object
-  ? { page: PageType; pageState: PageState }
-  : { page: PageType };
+  ? { page: PageType; pageState: PageState; excludeBasePath?: boolean }
+  : { page: PageType; excludeBasePath?: boolean };
```

Comment:
We should try to avoid nested ternary operators (https://github.com/elastic/kibana/blob/7.9/STYLEGUIDE.md#only-use-ternary-operators-for-small-simple-code)

Reply:
As of now I'm not sure what's a good replacement for this. I experimented with a few ways but this way as of right now is the most explicit type that covers all the requirements.